py/desitarget/mock/build.py

```diff
@@ -855,47 +855,40 @@ def _update_spectra_status(result):
                 log.info('No {} standards on healpix {}, {} not written.'.format(stdsuffix, pixnum, stdfile))
 
         # Finally write out the rest of the targets.
-        # targetsfile = os.path.join(output_dir, 'targets-{}.fits'.format(healsuffix))
-        # truthfile = os.path.join(output_dir, 'truth-{}.fits'.format(healsuffix))
-        # truthspecfile = os.path.join(output_dir, 'spectra-truth-{}.fits'.format(healsuffix))
-
         targetsfile = mockio.findfile('targets', nside, pixnum, basedir=output_dir)
         truthfile = mockio.findfile('truth', nside, pixnum, basedir=output_dir)
-        truthspecfile = mockio.findfile('spectra_truth', nside, pixnum, basedir=output_dir)
 
         inpixel = (pixnum == targpix)
         npixtargets = np.count_nonzero(inpixel)
         if npixtargets > 0:
             log.info('Writing {} targets to {}'.format(npixtargets, targetsfile))
+            targets.meta['EXTNAME'] = 'TARGETS'
             try:
                 targets[inpixel].write(targetsfile+'.tmp', format='fits', overwrite=True)
             except:
                 targets[inpixel].write(targetsfile+'.tmp', format='fits', clobber=True)
             os.rename(targetsfile+'.tmp', targetsfile)
-        
+
             log.info('Writing {}'.format(truthfile))
-            try:
-                truth[inpixel].write(truthfile+'.tmp', format='fits', overwrite=True)
-            except:
-                truth[inpixel].write(truthfile+'.tmp', format='fits', clobber=True)
-            os.rename(truthfile+'.tmp', truthfile)
-        
-            log.info('Writing {}'.format(truthspecfile))
             hx = fits.HDUList()
+            hdu = fits.convenience.table_to_hdu(truth[inpixel])
+            hdu.header['EXTNAME'] = 'TRUTH'
+            hx.append(hdu)
+
             hdu = fits.ImageHDU(Spectra.wave.astype(np.float32), name='WAVE', header=truthhdr)
             hdu.header['BUNIT'] = 'Angstrom'
             hdu.header['AIRORVAC'] = 'vac'
             hx.append(hdu)
-        
+
             hdu = fits.ImageHDU(trueflux[inpixel, :].astype(np.float32), name='FLUX')
             hdu.header['BUNIT'] = '1e-17 erg/s/cm2/Angstrom'
             hx.append(hdu)
-            
+
             try:
-                hx.writeto(truthspecfile+'.tmp', overwrite=True)
+                hx.writeto(truthfile+'.tmp', overwrite=True)
             except:
-                hx.writeto(truthspecfile+'.tmp', clobber=True)
-            os.rename(truthspecfile+'.tmp', truthspecfile)
+                hx.writeto(truthfile+'.tmp', clobber=True)
+            os.rename(truthfile+'.tmp', truthfile)
 
 def join_targets_truth(output_dir='.', nside=64, healpixels=None, verbose=True):
     """Combine all the target and truth catalogs generated by targets_truth into a
```

Comment:
Thanks for the quick review and for catching the replication.  I refactored the script functionality into this module and structured it so that MPI is now optional.  

Normal serial:
```
join_mock_targets [options]
```
With 4 MPI ranks reading in parallel:
```
srun -n 4 join_mock_targets [options] --mpi
```